helm/config.go

```diff
@@ -153,19 +154,56 @@ type Node interface {
 	write(enc *Encoder, prefix string)
 }
 
+// NewNode creates a new scalar node.
+func NewNode(value interface{}, modifiers ...NodeModifier) Node {
+	var node Node
+	if value == nil {
+		node = &Scalar{value: "~"}
+	} else if _, ok := value.(Node); ok {
+		node = value.(Node)
+	} else {
+		switch reflect.TypeOf(value).Kind() {
+		case reflect.Map:
+			valueOf := reflect.ValueOf(value)
+			mapping := NewEmptyMapping()
+			for _, key := range valueOf.MapKeys() {
+				mapping.Add(key.Interface().(string), valueOf.MapIndex(key).Interface())
```

Comment:
Yes, it will fail if the key in the YAML is not a string.  That shouldn't happen if the definition in the role manifest itself is valid, as I don't think kube uses non-string keys.

But what should we do? Catch the type mis-match and call panic ourselves? That doesn't seem better than what we get by default.

Reply:
I was hoping we'd be able to propagate the error up with enough information for the user to figure out where they messed things up and fix it.  But that was before I realized that this function had no error returning capability…